File: app/src/main/java/org/test/kotlin/demo/api/GitHubModule.java
package org.test.kotlin.demo.api;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public interface GitHubModule {

    @Provides
    static GitHubAPI provideAPI(Retrofit.Builder retrofit) {
        return retrofit
                .baseUrl("https://api.github.com")
                .build()
                .create(GitHubAPI.class);
    }

}

File: app/src/main/java/org/test/kotlin/demo/api/GitHubModule.kt
package org.test.kotlin.demo.api

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Provider
import javax.inject.Singleton

@Module
abstract class GitHubModule {

    @Module
    companion object {

        @Provides
        @Singleton
        @JvmStatic
        fun provideAPI(retrofit: Provider<Retrofit.Builder>) = retrofit
                .get()
                .baseUrl("https://api.github.com/")
                .build()
                .create(GitHubAPI::class.java)!!

    }

}
